### PR TITLE
[Openuv] Providing an iconserver

### PR DESCRIPTION
**Pull request by clinique**
Included some small Java 17 code improvement + addition of an icon server providing dedicated dynamic icons.

### CODE REVIEW

**Comment by lolodomo** [comment]
A fan of this new feature ;)
Will try to find the time to review it.

**Comment by clinique** [comment]
Yes, I think it greatly improves user experience when out of the box you've got some adapted icons and will not have to go chase them on Internet.

**Comment by lolodomo** [comment]
@clinique : there is no copyright on the icons you are providing ?

**Comment by clinique** [comment]
> @clinique : there is no copyright on the icons you are providing ?

Generic (ozone, UV alert) come from SVG Repo (open license) and have been adapted by myself. The UV Index icons (1 to 11) are the standard found everywhere used to represent level and color.

**Comment by lolodomo** [comment]
> I think it greatly improves user experience when out of the box you've got some adapted icons and will not have to go chase them on Internet.

I fully agree.

**Comment by clinique** [comment]
From what I see [here](https://www.who.int/publications/m/item/the-uv-index-sun-protection-table), these logos can be used free of charge.

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java [hunk 1]
```diff
@@ -158,28 +148,26 @@ private void updateChannels(ReportConfiguration config) {
...
-        if (refresh != null && !refresh.isCancelled()) {
-            refresh.cancel(true);
-        }
-        refreshJob = null;
 
-        ScheduledFuture<?> uxMax = uvMaxJob;
-        if (uxMax != null && !uxMax.isCancelled()) {
-            uxMax.cancel(true);
+        cancelFuture(refreshJob);
+        cancelFuture(uvMaxJob);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java:153]: You could `refreshJob`  and `uvMaxJob` to null just after.

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java [hunk 2]
```diff
@@ -158,28 +148,26 @@ private void updateChannels(ReportConfiguration config) {
...
+    }
+
+    private void cancelFuture(@Nullable ScheduledFuture<?> job) {
+        if (job != null && !job.isCancelled()) {
+            job.cancel(true);
         }
-        uvMaxJob = null;
+        job = null;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java:160]: See my other comment: to be done rather by the caller.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return getIconSets(null);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java:70]: Done

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java:153]: Done

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java (hunk 2 above)
Comment (clinique) [reply bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java:160]: Done

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by jlaur** [comment]
@clinique - can we rephrase PR title to something user-oriented?

**Pull request merged**